extensions/community/OneSignal.json

```
@@ -0,0 +1,221 @@
+  "name": "OneSignal",
+  "previewIconUrl": "https://asset-resources.gdevelop.io/public-resources/Icons/Line Hero Pack/Master/SVG/Communication/83c3b1f43896ce61930a58332b1fa00a1dcd9d1cabb5eec469ea54b30f173cba_Communication_chat_message_sms_bell_ring_notification.svg",
+  "shortDescription": "OneSignal is the fastest and most reliable service to send push notifications, in-app messages, SMS, and emails.",
+  "version": "1.0.0",
+  "description": "Check out the documentation here - https://documentation.onesignal.com/docs/onesignal-platform",
+  "tags": [
+    "network"
+  ],
+  "authorIds": [
+    "jy7FXnGX0ZZcWfrAI9YuQaeIphi1"
+  ],
+  "dependencies": [
+    {
+      "exportName": "onesignal-cordova-plugin",
+      "name": "onesignal-cordova-plugin",
+      "type": "cordova",
+      "version": "3.3.1"
+    }
+  ],
+  "eventsFunctions": [
+    {
+      "description": "Initialize OneSignal.",
+      "fullName": "Initialize OneSignal",
+      "functionType": "Action",
+      "name": "Initialize",
+      "sentence": "Initialize OneSignal",
+      "events": [
+        {
+          "type": "BuiltinCommonInstructions::JsCode",
+          "inlineCode": [
+            "const appid = eventsFunctionContext.getArgument(\"appid\");",
+            "const prompt = eventsFunctionContext.getArgument(\"prompt\");",
+            "const permission = eventsFunctionContext.getArgument(\"permission\");",
+            "",
+            "document.addEventListener('deviceready', OneSignalInit, false);",
+            "function OneSignalInit() {",
+            "    // NOTE: Update the init value below with your OneSignal AppId.",
+            "    window.plugins.OneSignal.init(appid);",
+            "    window.plugins.OneSignal.Notifications.setNotificationOpenedHandler(function(opened) {",
+            "        let notificationData = JSON.stringify(opened);",
+            "        console.log('notificationOpenedCallback: ' + notificationData);",
+            "    });",
+            "",
+            "    if (prompt == true) {",
+            "        window.plugins.OneSignal.Notifications.requestPermission(function(accepted) {",
+            "        console.log(\"User accepted notifications: \" + accepted);",
+            "        permission = accepted",
+            "    });",
+            "    }",
+            "    ",
+            "}"
+          ],
+          "parameterObjects": "",
+          "useStrict": true,
+          "eventsSheetExpanded": false
+        }
+      ],
+      "parameters": [
+        {
+          "description": "Your onesignal app id",
+          "name": "appid",
+          "type": "string"
+        },
+        {
+          "defaultValue": "yes",
+          "description": "Prompts the user for notification permissions",
```

```suggestion
          "description": "Prompt the user for notification permissions?",
```
It's a yes or no question, let's formulate it as such 🙂